paint-prototype/Assets/Fork/Tom/Scripts/PlayerMovement.cs

```diff
@@ -0,0 +1,281 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerMovement : MonoBehaviour
+{
+    [SerializeField, Range(0f, 100f)]
+    float maxSpeed = 10f;
+
+    [SerializeField, Range(0f, 100f)]
+    float maxAcceleration = 10f, maxAirAcceleration = 1f;
+
+    [SerializeField, Range(0f, 10f)]
+    float jumpHeight = 3f;
+
+    [SerializeField, Range(0, 5)]
+    int maxAirJumps = 0;
+
+    [SerializeField, Range(0, 90)]
+    float maxGroundAngle = 25f, maxStairsAngle = 50f;
+
+    [SerializeField, Range(0f, 100f)]
+    float maxSnapSpeed = 100f;
+
+    [SerializeField, Min(0f)]
+    float probeDistance = 1f;
+
+    [SerializeField]
+    LayerMask probeMask = -1, stairsMask = -1;
+
+    Rigidbody rBody;
```

Comment:
Naming should be more descriptive, inside a little loop it is okay to short-name the variables, but for this big of a scope it should have more descriptive names.